lib/auth/methods.go

```diff
@@ -465,16 +465,24 @@ func (s *Server) AuthenticateSSHUser(ctx context.Context, req AuthenticateSSHReq
 		return nil, trace.Wrap(err)
 	}
 
-	// Return the host CA for this cluster only.
-	authority, err := s.GetCertAuthority(ctx, types.CertAuthID{
-		Type:       types.HostCA,
-		DomainName: clusterName.GetClusterName(),
-	}, false)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	hostCertAuthorities := []types.CertAuthority{
-		authority,
+	var hostCertAuthorities []types.CertAuthority
+	if s.sendAllHostCAs {
+		hostCertAuthorities, err = s.GetCertAuthorities(ctx, types.HostCA, false)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	} else {
+		// Return the host CA for this cluster only.
+		authority, err := s.GetCertAuthority(ctx, types.CertAuthID{
+			Type:       types.HostCA,
+			DomainName: clusterName.GetClusterName(),
+		}, false)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		hostCertAuthorities = []types.CertAuthority{
+			authority,
+		}
```

Comment:
Loading all certs in server side and returning them to the client can affect not only ssh flow. 
What would you say to moving this logic to tsh and in case of `sendAllHostCAs` cluster setting the client will decide during tsh ssh flow if all cas should be loaded ?